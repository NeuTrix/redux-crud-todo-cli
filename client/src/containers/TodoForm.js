import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Row, Col, Button, Form } from 'react-bootstrap';

import TodoTitleBar from './TodoTitleBar';
// import CalendarBtn from './CalendarBtn';

// ========= ========= ========= 
const TodoForm = (props) => {

	// create a default format for today's date
	const getDay = (date) => {
		let year = date.getFullYear();
		let month = date.getMonth()+1;
		let day = date.getDate();
		let fullDate =[];
		return [...fullDate, year, month, day].join('-');
	};
	const _today = getDay(new Date());

	let _task, _rank, _date;
	
	const handleSubmit = (event) => {
		event.preventDefault();
		props.addTodo({
			date: _date.value,
			task: _task.value, 
			rank: _rank.value, 
		});
	
		// reset the fields
		_task.value ='';
		_task.focus();
	};

	const styleTop ={
		backgroundColor: 'lightblue',
		borderRadius: 5,
		padding:10,
		marginBottom: 20,
	};

	return (
		<Grid className= 'TodoForm'>

			<Form onSubmit={ handleSubmit }>

				<Row className= 'AddTodoTitle'>	
					<h3>
						Add a new Todo here:
					</h3>
				</Row>

				<Row>	
					<TodoTitleBar/>
				</Row>

				<Row style= { styleTop }>

					<Col sm = { 1 } > 
						<p>
							===>
						</p>
					</Col>

					<Col className='rank' sm = { 1 } > 

						<select  
							ref= { (value) => _rank = value }
							defaultValue='Med' 
						>
							<option value='High'>High</option>
							<option value='Med' >Med</option>
							<option value='Low'>Low</option>

						</select>

					</Col>

					<Col className='date' 
						sm = { 1 } 
					> 

						<input 
							ref= { (input) => _date = input } 
							type = 'date'  
							defaultValue = { _today } 
							required
						/> 

					</Col>

					<Col className='task' sm = { 7 } > 

						<input 
							ref= { (input) => _task = input } 
							type = 'text'  
							placeholder='enter a new task here...' 
							size={ 55 } 
							maxLength= {48} 
							required
						/>

					</Col>
			
					<Col className='reset' sm = { 1 } > 
						<Row>
							<Col sm= { 6 }>
								<Button 
									className= { 'btn btn-info btn-sm' } 
									type= 'submit' 
									value= 'Add' 
								> Add </Button>
							</Col>

							<Col sm= { 6 }>
								<Button 
									className= { 'btn btn-warning btn-sm' } 
									type = 'reset'
								> Reset 
								</Button> 
							</Col>

						</Row>
					</Col>

				</Row>

			</Form>

		</Grid>
	);
};

// ========= ========= ========= 
TodoForm.propTypes = {
	addTodo: PropTypes.func.isRequired,
	_task: PropTypes.string.isRequired,
	_rank: PropTypes.string.isRequired,
	_date: PropTypes.string.isRequired,
};

TodoForm.defaultProps = {
	addTodo: f=>f,
	_task: '',
	_rank: '',
	_date: ''
};
// ========= ========= ========= 

export default TodoForm;