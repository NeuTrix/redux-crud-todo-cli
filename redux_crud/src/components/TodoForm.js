import React { Component } from 'react'
import PropTypes from 'prop-types'

export default class TodoForm extends Component {

	render () {

		return (
			<div>
				<h1> The form will go here </h1>
			</div>
			)
	}
}
