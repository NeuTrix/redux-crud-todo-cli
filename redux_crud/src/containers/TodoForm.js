import React from 'react';
import PropTypes from 'prop-types';
// import { Form } from 'react-bootstrap';

const TodoForm = (props) => {

	let _task, _rank, _date
	
	const handleSubmit = (event) => {
		event.preventDefault();
		props.addTodo(_task.value, _rank.value, _date.value);
	
		// reset the fields
		_task.value ="";
		_task.focus();
	}

	return (
		<div>
			<h3> Todays Date:  {props._today} </h3>
			<h4> Add a new todo here: </h4>
			<form onSubmit={handleSubmit}>

				<select ref= {(input) => _rank = input}>
				  <option value="High">High</option>
				  <option value="Med" selected="selected">Med</option>
				  <option value="low">Low</option>
				</select>

				<input ref= {(input) => _task = input} type = "text"  placeholder="Task" size={40} required/>
				<input ref= {(input) => _date = input} type = "date"  defaultValue = {props._today} required/>
				
				<input type="submit"/>
				<input type = "reset"/>
			</form>
		</div>
	)
}

TodoForm.propTypes = {
	_task: PropTypes.string.isRequired,
	_rank: PropTypes.string.isRequired,
	_date: PropTypes.string.isRequired,
}

TodoForm.defaultProps = {
	_task: "",
	_rank: "",
	_date: ""
}

export default TodoForm