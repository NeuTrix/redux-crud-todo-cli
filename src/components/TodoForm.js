// ... Component to create new Todo items
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import normalizeDate from '../helpers/normalizeDate';

// +++++++++ CSS  +++++++++ 

const gridStyle = {
	display: 'grid',
	gridTemplateAreas: 
	` "task task task " 
		" priority date action" ` ,
	gridTemplateColumn: '30px 30px 30px',
	gridTemplateRow: 50,
	gridGap: 5,

	backgroundColor: 'transparent',
	padding: 5,
	border: '2px solid lime',
	borderRadius: 5,
}

const placement = {
	task: { gridArea: 'task' },
	action: { gridArea: 'action' },
	priority: { gridArea: 'priority' },
	date: { gridArea: 'date' },
}

// +++++++++ COMPONENT +++++++++ 

class TodoForm extends Component {

	constructor (props) {

		super(props);

		this.state = {
			date: normalizeDate(new Date()) ,
			task: '', 
			rank: 'Med',
		 	owner: this.props.owner ,
		}
		this.handleSubmit = this.handleSubmit.bind(this)
		this.handleChange = this.handleChange.bind(this)
	}

	handleSubmit(e) {
		e.preventDefault ();
		this.props.createTodo (this.state);
		this.setState ({ task: ''})
	};

	handleChange(e) {
		e.preventDefault();
		this.setState ({ [ e.target.name ]: e.target.value })
	}

	render () {
		return (
			<form 
				className = 'TodoForm' 
				style = { gridStyle } 
				onSubmit = { this.handleSubmit } 
			>
				<input 
					id = 'new_item_task'
					style = { placement.task }
					type = 'text'
					name = 'task'
					value = { this.state.task }
					onChange = { this.handleChange }
					placeholder = 'enter a new task here'
					autoFocus
					required
				/>

				<select
					id = 'new_item_priority' 
					style = { placement.priority } 
					name = 'rank'
					type = 'select'
					value = { this.state.rank }
					onChange = { this.handleChange }
			 	> 
					<option value = 'High'> High	</option>
					<option value = 'Med'>	Med		</option>
					<option value = 'Low'>	Low		</option>
				</select>

				<input 
					id = 'new_item_date'
					style = { placement.date } 
					name = 'date' 
					type = 'date'
					onChange = { this.handleChange }
					defaultValue = { this.state.date }
				/>

				<button 
					id = 'new_item_submit'
					style = { placement.action } 
					type = "Submit"
				> Add </button> 

			</form>
		)
	}
};

// +++++++++ PROPS +++++++++ 

TodoForm.propTypes = {
	// addTodo: PropTypes.func.isRequired,
	createTodo: PropTypes.func.isRequired,
	owner: PropTypes.string.isRequired
};

TodoForm.defaultProps = {
	// addTodo: f => f,
	createTodo: f => f,
	owner: 'Default from APP.js'
};

export default TodoForm;