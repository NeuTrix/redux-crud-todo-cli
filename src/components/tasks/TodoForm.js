// ... Component to create new Todo items
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import normalizeDate from '../helpers/normalizeDate';

// +++++++++ CSS  +++++++++ 

const gridStyle = {
	display: 'grid',
	gridTemplateAreas: 
	` "task task task " 
		" priority date action" ` ,
	gridTemplateColumn: '30px 30px 30px',
	gridTemplateRow: 50,
	gridGap: 5,
	borderColor: 'steelblue',
	padding: 10,
	marginBottom: 20
}

const style = {
	task: { 
		gridArea: 'task', 
		paddingLeft: '1em', 
	},

	action: { 
		gridArea: 'action', 
		backgroundColor: 'whitesmoke', 
		padding: 5,
		border: '1px solid darkgreen' ,
		color: 'palegreen',
		opacity: .8,
		alignContent: 'center',
	},

	priority: { 
		gridArea: 'priority' ,
		backgroundColor: 'aliceblue', 
		// // border: '1px solid steelblue' ,
		color: 'steelblue',
		opacity: .8,
	},

	date: { 
		gridArea: 'date' ,
		backgroundColor: 'aliceblue', 
		border: '1px solid steelblue' ,
		color: 'steelblue',
		opacity: .8,
	},
}

// +++++++++ COMPONENT +++++++++ 

class TodoForm extends Component {

	constructor (props) {

		super(props);

		this.state = {
			date: normalizeDate(new Date()) ,
			task: '', 
			rank: 'Med',
		 	owner: this.props.owner ,
		}
		this.handleSubmit = this.handleSubmit.bind(this)
		this.handleChange = this.handleChange.bind(this)
	}

	handleSubmit(e) {
		e.preventDefault ();
		this.props.createTodo (this.state);
		this.setState ({ task: ''})
	};

	handleChange(e) {
		e.preventDefault();
		this.setState ({ [ e.target.name ]: e.target.value })
	}

	render () {
		return (
			<form 
				className = 'TodoForm boxClr engrBox' 
				style = { gridStyle } 
				onSubmit = { this.handleSubmit } 
			>
				<input 
					id = 'new_item_task'
					style = { style.task }
					type = 'text'
					name = 'task'
					value = { this.state.task }
					onChange = { this.handleChange }
					placeholder = 'enter a new task here'
					autoFocus
					required
				/> 

				<select
					className = 'mat'
					id = 'new_item_priority' 
					style = { style.priority } 
					name = 'rank'
					type = 'select'
					value = { this.state.rank }
					onChange = { this.handleChange }
			 	> 
					<option value = 'High'> High	</option>
					<option value = 'Med'>	Med		</option>
					<option value = 'Low'>	Low		</option>
					
				</select>

				<input 
					className = 'mat'
					id = 'new_item_date'
					style = { style.date } 
					name = 'date' 
					type = 'date'
					onChange = { this.handleChange }
					value = { this.state.date }
				/>

				<button 
					style = { style.action }
					className = 'ctr clrBox mat'
					id = 'new_item_submit'
					type = 'submit'
				> <span 
					style = {{ color: 'palegreen' }}
					className = "ctr engr fa fa-plus fa-lg" 
					></span> 
				</button> 

			</form>
		)
	}
};

// +++++++++ PROPS +++++++++ 

TodoForm.propTypes = {
	createTodo: PropTypes.func.isRequired,
	owner: PropTypes.string.isRequired
};

TodoForm.defaultProps = {
	createTodo: f => f,
	owner: 'Default from APP.js'
};

export default TodoForm;