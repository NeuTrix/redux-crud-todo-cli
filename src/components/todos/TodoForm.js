import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components'
//  custom
import normalizeDate from '../../helpers/normalizeDate';
import { colors, media } from '../../helpers/cssConstants';


// +++++++++ CSS  +++++++++ 

const Grid = styled.form `
	/* mobile view */
	display: grid;
	grid-template-areas:
		" glyph task  task  task "
		" ...	 	rank 	date 	add " 
	;
	grid-template-columns: 1fr 3fr 4fr 1fr;
	grid-gap: 10px;

	background-color: aliceblue;
	padding: 10px;
	margin-bottom: 20px;

	& * {
		font-size: 1em;
		&:hover {
			color: lime;
		}
	}

	/* iPad and large view */
	@media (${media._medium}) {
		grid-template-areas: 
			" ...  	...   ...  	...   ...  "
			" glyph 	task  rank  date 	add  "
			" ...  	...   ...  	...   ...  "
		;
		grid-template-columns: 1fr 8fr 2fr 3fr 1fr;
	}
`;

const Add = styled.button `
	grid-area: add;
	color: #00cc00;
	background: none;
  border: none;
	font-size: 2em;
	`;

const DatePick = styled.input `
	grid-area: date;
	text-indent: 10px;
`;

const Rank = styled.select `
	grid-area: rank;
`;

const Task = styled.input `
	grid-area: task;
	color: steelblue;
  text-indent: 10px;
`;

const Glyph = styled.i `
	grid-area: glyph;
	color: ${colors._iceblue} ;
	font-size: 1.5em;
`;


// +++++++++ COMPONENT +++++++++ 

class TodoForm extends Component {

	constructor (props) {
		super(props);
		this.state = {
			date: normalizeDate(new Date()) ,
			task: '', 
			rank: 'Med',
		 	owner: this.props.owner ,
		}
		this.handleSubmit = this.handleSubmit.bind(this)
		this.handleChange = this.handleChange.bind(this)
	}

	handleSubmit(e) {
		e.preventDefault ();
		this.props.createTodo (this.state);
		this.setState ({ task: ''})
	};

	handleChange(e) {
		e.preventDefault();
		this.setState ({ [ e.target.name ]: e.target.value })
	}

	render () {

		return (
			<Grid 
				className= { `TodoForm ${ this.props.className } 
				boxClr engrBox` } 
				onSubmit= { this.handleSubmit } 
			>
				<Glyph className = 'ctr engr fa fa-tasks fa-lg'/>

				<Task 
					id= 'new_item_task'
					type= 'text'
					name= 'task'
					value= { this.state.task }
					onChange= { this.handleChange }
					placeholder= 'Enter a new task here'
					autoFocus
					required
				/> 

				<Rank
					className= 'mat'
					id= 'new_item_priority' 
					name= 'rank'
					type= 'select'
					value= { this.state.rank }
					onChange= { this.handleChange }
			 	> 
					<option value= 'High'> High	</option>
					<option value= 'Med'>	Med		</option>
					<option value= 'Low'>	Low		</option>
					
				</Rank>

				<DatePick 
					className= 'mat'
					id= 'new_item_date'
					name= 'date' 
					type= 'date'
					onChange= { this.handleChange }
					value= { this.state.date }
				/>

				<Add 
					className= 'ctr engr fa fa-plus fa-2x'
					id= 'new_item_submit'
					type= 'submit' 
				> 
				</Add> 

			</Grid>
		)
	}
};

// +++++++++ PROPS +++++++++ 

TodoForm.propTypes = {
	createTodo: PropTypes.func.isRequired,
	owner: PropTypes.string.isRequired
};

TodoForm.defaultProps = {
	createTodo: f => f,
	owner: 'Default from APP.js'
};

export default TodoForm;