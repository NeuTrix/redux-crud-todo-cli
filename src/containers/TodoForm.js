import React from 'react';
import PropTypes from 'prop-types';
import TodoTitleBar from './TodoTitleBar';
import normalizeDate from '../myFunctions/normalizeDate';

import { 
	Grid, 
	Row, 
	Col, 
	Button, 
	Form, 
	FormControl 
} from 'react-bootstrap';

// ========= Component

const TodoForm = (props) => {

	let _currentDate = normalizeDate(new Date());
	let _task, // todo task
		_rank, // todo priority
		_date; // todo due date

	const handleSubmit = (event) => {
		event.preventDefault();
		props.addTodo({
			date: _date.value,
			task: _task.value, 
			rank: _rank.value, 
		});
		_task.value = ''; // reset the fields
		_task.focus();
	};

	const styleTop = {
		backgroundColor: 'lightblue',
		borderRadius: 5,
		padding:10,
		marginBottom: 20,
	};

	const spacing = [1, 1, 2, 7, 1] // control spacing of elements

	return (
		<Grid className = 'TodoForm'>

			<Form onSubmit = { handleSubmit }>

				<Row className = 'AddTodoTitle'>	
					<h3> Add a new Todo here: </h3>
				</Row>

				<Row>	
					<Col xsHidden = 'true' smHidden = 'true'>
						<TodoTitleBar/>
					</Col>
				</Row>

				<Row style= { styleTop }>

					<Col className ='reset' sm = { spacing[0] } > 
						<Button 
							className = { 'btn btn-warning btn-sm' } 
							type = 'reset'> 
								Reset 
						</Button> 
					</Col>

					<Col className ='rank' sm = { spacing[1] } > 
						<FormControl 
							bsSize = 'sm'
							defaultValue = 'M'
							componentClass = 'select' 
							inputRef = { (value) => { _rank = value } } >

							<option value = 'H'>H</option>
							<option value = 'M'>M</option>
							<option value = 'L'>L</option>

			      </FormControl>
					</Col>

					<Col className ='date'	sm = { spacing[2] } >
						<FormControl 
							inputRef = { (input) => { _date = input } } 
							type = 'date'
							defaultValue = { _currentDate } 
							bsSize = 'sm'
							required /> 
					</Col>

					<Col className ='task' sm = { spacing[3] } > 
						<FormControl 
							inputRef = { (input) => { _task = input;} } 
							type = 'text'  
							placeholder ='enter a new task here...' 
							size = { 55 } 
							maxLength = { 48 } 
							required />
					</Col>
			
					<Col className = 'addBtn' sm = { spacing[4] } > 
						<Button 
							className = { 'btn btn-success btn-sm' } 
							type = 'submit' 
							value = 'Add' > 
								Add 
						</Button>
					</Col>

				</Row>
			</Form>
		</Grid>
	);
};

// ========= Props  

TodoForm.propTypes = {
	addTodo: PropTypes.func.isRequired,
	_task: PropTypes.string.isRequired,
	_rank: PropTypes.string.isRequired,
	_date: PropTypes.string.isRequired
};

TodoForm.defaultProps = {
	addTodo: f => f,
	_task: '',
	_rank: '',
	_date: '2020-12-31'
};

export default TodoForm;