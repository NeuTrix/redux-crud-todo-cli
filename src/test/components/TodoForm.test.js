/* eslint-env node, mocha */
import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'enzyme';
import TodoForm from '../../components/todos/TodoForm';

xit('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(
		<TodoForm
			createTodo={f => f}
			owner="100"
		/>, div,
	);
});

const wrapper = render(
	<TodoForm
		createTodo={f => f}
		owner = "100"
	/>,
);

xit('...renders the TodoForm ', () => {
	expect(wrapper).toMatchSnapshot();
});
