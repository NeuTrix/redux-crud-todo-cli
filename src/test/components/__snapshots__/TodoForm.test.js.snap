// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`...renders the TodoForm  1`] = `
<form
  class="MuiFormControl-root-3 TodoForm-grid-2"
>
  <div
    class="MuiFormControl-root-3 MuiFormControl-marginDense-5 MuiFormControl-fullWidth-6"
    style="grid-area:task"
  >
    <label
      class="MuiFormLabel-root-18 MuiFormLabel-required-23 MuiInputLabel-required-11 MuiInputLabel-root-7 MuiInputLabel-formControl-12 MuiInputLabel-animated-15 MuiInputLabel-marginDense-13 MuiInputLabel-outlined-17"
      data-shrink="false"
    >
      enter new task
      <span
        class="MuiFormLabel-asterisk-24"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root-38 MuiOutlinedInput-root-25 MuiInputBase-fullWidth-47 MuiInputBase-formControl-39 MuiInputBase-marginDense-45"
    >
      <fieldset
        aria-hidden="true"
        class="MuiPrivateNotchedOutline-root-55 MuiOutlinedInput-notchedOutline-32"
        style="padding-left:8px"
      >
        <legend
          class="MuiPrivateNotchedOutline-legend-56"
          style="width:0.01px"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
      <input
        aria-invalid="false"
        class="MuiInputBase-input-48 MuiOutlinedInput-input-33 MuiInputBase-inputMarginDense-49 MuiOutlinedInput-inputMarginDense-34"
        name="task"
        required=""
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="MuiFormControl-root-3 MuiFormControl-marginDense-5 MuiFormControl-fullWidth-6"
    style="grid-area:rank"
  >
    <label
      class="MuiFormLabel-root-18 MuiFormLabel-filled-22 MuiInputLabel-root-7 MuiInputLabel-formControl-12 MuiInputLabel-animated-15 MuiInputLabel-shrink-14 MuiInputLabel-marginDense-13 MuiInputLabel-outlined-17 Rank-label-57"
      data-shrink="true"
      for="new_item_priority"
    >
      rank
    </label>
    <div
      class="MuiInputBase-root-38 MuiOutlinedInput-root-25 MuiInputBase-fullWidth-47 MuiInputBase-formControl-39 MuiInputBase-marginDense-45"
    >
      <fieldset
        aria-hidden="true"
        class="MuiPrivateNotchedOutline-root-55 MuiOutlinedInput-notchedOutline-32"
        style="padding-left:8px"
      >
        <legend
          class="MuiPrivateNotchedOutline-legend-56"
          style="width:0"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
      <div
        class="MuiSelect-root-60"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          class="MuiSelect-select-61 MuiSelect-selectMenu-64 MuiSelect-outlined-63 MuiInputBase-input-48 MuiOutlinedInput-input-33 MuiInputBase-inputMarginDense-49 MuiOutlinedInput-inputMarginDense-34"
          role="button"
          tabindex="0"
        >
          Med
        </div>
        <input
          id="new_item_priority"
          name="rank"
          type="hidden"
          value="Med"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-67 MuiSelect-icon-66"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="MuiFormControl-root-3 MuiFormControl-marginDense-5 MuiFormControl-fullWidth-6"
    style="grid-area:date"
  >
    <label
      class="MuiFormLabel-root-18 MuiFormLabel-filled-22 MuiInputLabel-root-7 MuiInputLabel-formControl-12 MuiInputLabel-animated-15 MuiInputLabel-shrink-14 MuiInputLabel-marginDense-13 MuiInputLabel-outlined-17"
      data-shrink="true"
    >
      set due date
    </label>
    <div
      class="MuiInputBase-root-38 MuiOutlinedInput-root-25 MuiInputBase-fullWidth-47 MuiInputBase-formControl-39 MuiInputBase-marginDense-45"
    >
      <fieldset
        aria-hidden="true"
        class="MuiPrivateNotchedOutline-root-55 MuiOutlinedInput-notchedOutline-32"
        style="padding-left:8px"
      >
        <legend
          class="MuiPrivateNotchedOutline-legend-56"
          style="width:0"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
      <input
        aria-invalid="false"
        class="MuiInputBase-input-48 MuiOutlinedInput-input-33 MuiInputBase-inputType-51 MuiInputBase-inputMarginDense-49 MuiOutlinedInput-inputMarginDense-34"
        name="date"
        type="date"
        value="2019-01-08"
      />
    </div>
  </div>
  <button
    class="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-contained-91 MuiButton-containedSecondary-93 MuiButton-raised-94 MuiButton-raisedSecondary-96 TodoForm-button-1"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label-81"
    >
      Add
    </span>
  </button>
</form>
`;
